src/ceph-daemon/ceph-daemon

```diff
@@ -1616,6 +1616,105 @@ def command_rm_cluster():
 
 ##################################
 
+def check_time_sync():
+    units = ['chronyd.service', 'systemd.timesyncd', 'ntpd.service']
+    for u in units:
+        (enabled, state) = check_unit(u)
+        if enabled and state == 'running':
+            logger.info('Time sync unit %s is enabled and running' % u)
+            return True
+    return False
+
+def check_firewalld(fix=False):
+    cmd = find_program('firewall-cmd')
+    if not cmd:
+        logger.info('firewalld does not appear to be present')
+        return True  # no firewalld -> all good
+    (enabled, state) = check_unit('firewalld.service')
+    if not enabled:
+        logger.info('firewalld.service is not enabled')
+        return True
+    for svc in ['ceph', 'ceph-mon']:
+        out, err, ret = call([cmd, '--permanent', '--query-service', svc])
```

Comment:
have you considered the firewall.client module instead of issuing CLI calls, and interpreting output? Either way we'll need to extend this approach to encompass the ports required for mgr modules like dashboard and prometheus. For example if we enable dashboard by default, and firewalld is active it'll block 8443

Reply:
We can't make use of any dependencies, as long as we have we want to support downloading `ceph-daemon` via wget.

Reply:
All of this firewalld stuff should only be done if firewalld is installed and the service is enabled.

Reply:
the base os install has a ceph.xml and ceph-mon.xml.. how do you think we should we deal with all of the other services?

Reply:
I don't see how using the firewall python library is a dependency - it's installed/required by  firewalld and underpins the firewall-cmd CLI. We could put the import in a try, and if it fails use that to determine that firewalld is unavailable (i.e. set a bool). The ceph-mon.xml and ceph.xml files are actually installed by firewalld, not the OS. The mechanics of opening/closing a port should probably be in a separate function so it can be called by the bootstrap, and then called during mgr module enable workflow of the install of the monitoring containers. Each service would then handle their own firewall config requirement but call the same execution code?

Reply:
Ok, this makes me think I pull the firewalld stuff out of this PR completely and treat it as part of the deploy process.